exist-core/src/test/xquery/xquery3/fnCollationKey.xqm

```diff
@@ -0,0 +1,112 @@
+(:
+ : eXist-db Open Source Native XML Database
+ : Copyright (C) 2001 The eXist-db Authors
+ :
+ : info@exist-db.org
+ : http://www.exist-db.org
+ :
+ : This library is free software; you can redistribute it and/or
+ : modify it under the terms of the GNU Lesser General Public
+ : License as published by the Free Software Foundation; either
+ : version 2.1 of the License, or (at your option) any later version.
+ :
+ : This library is distributed in the hope that it will be useful,
+ : but WITHOUT ANY WARRANTY; without even the implied warranty of
+ : MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ : Lesser General Public License for more details.
+ :
+ : You should have received a copy of the GNU Lesser General Public
+ : License along with this library; if not, write to the Free Software
+ : Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ :)
+xquery version "3.1";
+
+module namespace fnck="http://exist-db.org/xquery/test/function_collation_key";
+
+declare namespace test="http://exist-db.org/xquery/xqsuite";
+
+declare
+    %test:assertTrue
+function fnck:default-equal-case() {
+    let $first := fn:collation-key("a")
+    let $second := fn:collation-key("a")
+    return $first = $second
```

Comment:
As you are trying to compare the equality of two Atomic Values, instead of the sequence equality comparison operator `=`, you should use the atomic value comparison operator `eq`.

See slide 30 here - https://www.adamretter.org.uk/presentations/xquery-and-xml-databases_xml-summer-school_oxford_20190919.pdf